perf/allocs_per_case.jl

```diff
@@ -1,6 +1,5 @@
 # Launch with `julia --project --track-allocation=user`
 import Pkg
-Pkg.develop(path = ".")
```

Comment:
Are we sure that this is safe? Won't this lead to analyzing the latest release and not the PR?

Reply:
This is where in incremental compilation issue is coming from.  I don't think we can run this concurrently with the other test cases so we'll have to run sequentially at the end.

Reply:
We already `dev` the ClimaCore package at the toplevel

Reply:
Yeah, but ReportMetrics.jl calls
```julia
run(`julia --project=perf perf/allocations_per_case.jl`)
```
And I think it's that is the session that needs the `Pkg.develop`

Reply:
I think the real fix for the incremental compilation issue if running in the same project do these analysis sequentially unfortunately , so I added back in the develop because it is nicer for running things locally.